src/moneyTracker.js

```diff
@@ -0,0 +1,35 @@
+function Moneytracker() {
+  this._transactions = [];
+};
+
+Moneytracker.prototype.recordEarning = function (money, date) {
+  this._transactions.push(new Transaction(money, date))
+};
+
+Moneytracker.prototype.recordExpenditure = function (money, date) {
+  this._transactions.push(new Transaction(money, date))
+};
+
+Moneytracker.prototype._balance = function () {
+  var totalBalance = 0;
+  for(i = 0; i < this._transactions.length; i++) {
```

Comment:
`i` is a global variable.  (Have a read about what the `var` keyword does.)  Try to avoid globals.